### PR TITLE
Add URL value to the CallSummary

### PR DESCRIPTION
**Pull request by jhford**
I think it's useful to know the URL that actually ended up being used in the
call.  This PR adds the URL to the CallSummary.  I think I added it to all of
the relevant places, but I'm not sure

### CODE REVIEW

File: http.go [hunk 1]
```diff
@@ -40,10 +40,14 @@ type CallSummary struct {
 	HTTPResponseBody string
 	// Keep a record of how many http requests were attempted
 	Attempts int
+	// Keep a copy of the URL that was used
+	URL string
+	// Keep a copy of the HTTP Method used
+	Method string
```
Comment (petemoore) [inline http.go]: These two pieces of data are already in `HTTPRequest` member of this struct, so we don't need to add them.

File: http.go [hunk 2]
```diff
@@ -40,10 +40,14 @@ type CallSummary struct {
...
 }
 
 func (cs *CallSummary) String() string {
-	return fmt.Sprintf("\nCALL SUMMARY\n============\nRequest Headers:\n%#v\nRequest Body:\n%v\nResponse Headers:\n%#v\nResponse Body:\n%v\nAttempts: %v", cs.HTTPRequest.Header, cs.HTTPRequestBody, cs.HTTPResponse.Header, cs.HTTPResponseBody, cs.Attempts)
+	return fmt.Sprintf("\nCALL SUMMARY\n============\n%s %s\nRequest Headers:\n%#v\nRequest Body:\n%v\nResponse Headers:\n%#v\nResponse Body:\n%v\nAttempts: %v", cs.Method, cs.URL, cs.HTTPRequest.Header, cs.HTTPRequestBody, cs.HTTPResponse.Header, cs.HTTPResponseBody, cs.Attempts)
```
Comment (petemoore) [inline http.go]: We can replace this with:

```
return fmt.Sprintf("\nCALL SUMMARY\n============\n%s %s\nRequest Headers:\n%#v\nRequest Body:\n%v\nResponse Headers:\n%#v\nResponse Body:\n%v\nAttempts: %v", cs.HTTPRequest.Method, cs.HTTPRequest.URL, cs.HTTPRequest.Header, cs.HTTPRequestBody, cs.HTTPResponse.Header, cs.HTTPResponseBody, cs.Attempts)
```

File: http.go [hunk 3]
```diff
@@ -84,6 +88,7 @@ func setURL(client *Client, route string, query url.Values) (u *url.URL, err err
 func (client *Client) Request(rawPayload []byte, method, route string, query url.Values) (*CallSummary, error) {
 	callSummary := new(CallSummary)
 	callSummary.HTTPRequestBody = string(rawPayload)
+	callSummary.Method = method
```
Comment (petemoore) [inline http.go]: Not needed (see above).

File: http.go [hunk 4]
```diff
@@ -92,6 +97,7 @@ func (client *Client) Request(rawPayload []byte, method, route string, query url
 		var ioReader io.Reader
 		ioReader = bytes.NewReader(rawPayload)
 		u, err := setURL(client, route, query)
+		callSummary.URL = u.String()
```
Comment (petemoore) [inline http.go]: Not needed (see above).

**Review by petemoore — CHANGES_REQUESTED** [review]
I tested my proposed alternative with:

```go
package main

import (
	"log"
	"os"

	"github.com/taskcluster/taskcluster-client-go"
	"github.com/taskcluster/taskcluster-client-go/secrets"
)

func main() {
	c := tcclient.Credentials{
		ClientID:    os.Getenv("TASKCLUSTER_CLIENT_ID"),
		AccessToken: os.Getenv("TASKCLUSTER_ACCESS_TOKEN"),
		Certificate: os.Getenv("TASKCLUSTER_CERTIFICATE"),
	}
	s := secrets.New(&c)
	_, err := s.Get("fred")
	if err != nil {
		switch e := err.(type) {
		case *tcclient.APICallException:
			log.Printf("%v", e.CallSummary)
		}
		os.Exit(1)
	}
}
```

This produced:

```
2017/10/23 20:19:37 
CALL SUMMARY
============
GET https://secrets.taskcluster.net/v1/secret/fred
Request Headers:
http.Header{"Authorization":[]string{"Hawk id=\"mozilla-ldap/pmoore@mozilla.com/dev\", mac=\"Wfkg3yN12604KKoqkKhMg7op1IoGax14Jlhd90sB3oI=\", ts=\"1508782775\", nonce=\"TIfuqlfQ\""}}
Request Body:

Response Headers:
http.Header{"Content-Security-Policy":[]string{"report-uri /__cspreport__;default-src 'none';frame-ancestors 'none';"}, "Access-Control-Allow-Origin":[]string{"*"}, "Access-Control-Request-Method":[]string{"*"}, "Access-Control-Allow-Headers":[]string{"X-Requested-With,Content-Type,Authorization,Accept,Origin"}, "Content-Length":[]string{"339"}, "Strict-Transport-Security":[]string{"max-age=7776000"}, "Via":[]string{"1.1 vegur"}, "Access-Control-Allow-Methods":[]string{"OPTIONS,GET,HEAD,POST,PUT,DELETE,TRACE,CONNECT"}, "Content-Type":[]string{"application/json; charset=utf-8"}, "Etag":[]string{"W/\"153-sopvjomGOVB3JJsolmKr1Q\""}, "Date":[]string{"Mon, 23 Oct 2017 18:19:37 GMT"}, "Server":[]string{"Cowboy"}, "Connection":[]string{"keep-alive"}}
Response Body:
{
  "code": "ResourceNotFound",
  "message": "Secret not found\n----\nmethod:     get\nerrorCode:  ResourceNotFound\nstatusCode: 404\ntime:       2017-10-23T18:19:37.231Z",
  "requestInfo": {
    "method": "get",
    "params": {
      "0": "fred",
      "name": "fred"
    },
    "payload": {},
    "time": "2017-10-23T18:19:37.231Z"
  }
}
Attempts: 1
exit status 1
```

File: http.go (hunk 1 above)
Comment (jhford) [reply http.go]: I didn't notice!  good call.

File: http.go (hunk 2 above)
Comment (jhford) [reply http.go]: Sure!  looks good to me

**Comment by jhford** [comment]
Well that certainly simplifies the PR :)

**Review by petemoore — APPROVED** [review]
Awesome, thanks John!

**Pull request merged**